### PR TITLE
feat: implement voice channel statuses

### PR DESCRIPTION
**Pull request by Icebluewolf**
## Summary

Implements voice channel statuses by adding the `VoiceChannel.status` property. Set the status with `VoiceChannel.set_status()`. And adds the `on_voice_channel_status_update` event. As of posting it does not implement the corresponding audit log actions. 

Related DAPI docs PRs: [#6398](https://github.com/discord/discord-api-docs/pull/6398) and [#6400](https://github.com/discord/discord-api-docs/pull/6400)

## Information

- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
- [x] If `type: ignore` comments were used, a comment is also left explaining why.
- [x] I have updated the changelog to include these changes.

### CODE REVIEW

**Review by plun1331 — CHANGES_REQUESTED** [review]
chsngelog

**Review by Lulalaby — CHANGES_REQUESTED** [review]
u have a typo in commit
amend :3

File: CHANGELOG.md [hunk 1]
```diff
@@ -93,6 +93,8 @@ These changes are available on the `master` branch, but have not yet been releas
   ([#2342](https://github.com/Pycord-Development/pycord/pull/2342))
 - Added `invitable` and `slowmode_delay` to `Thread` creation methods.
   ([#2350](https://github.com/Pycord-Development/pycord/pull/2350))
+- Added support for voice channel statuses
```
Comment (plun1331) [inline CHANGELOG.md:96]:
```suggestion
- Added support for voice channel statuses.
```

File: discord/channel.py [hunk 2]
```diff
@@ -1955,6 +1970,31 @@ async def create_activity_invite(
             **kwargs,
         )
 
+    async def set_status(self, status: str | None, *, reason=None) -> None:
```
Comment (plun1331) [inline discord/channel.py:1973]: reason typehint
```suggestion
    async def set_status(self, status: str | None, *, reason: str | None = None) -> None:
```

File: discord/http.py [hunk 3]
```diff
@@ -2185,6 +2185,12 @@ def move_member(
             guild_id=guild_id, user_id=user_id, channel_id=channel_id, reason=reason
         )
 
+    def set_voice_channel_status(
+        self, channel_id: Snowflake, payload: Any, *, reason: str | None = None
```
Comment (plun1331) [inline discord/http.py:2189]: I would revise this to take `status` as a string param and build the payload, similar to the other http methods

File: discord/state.py [hunk 4]
```diff
@@ -1785,6 +1785,28 @@ def parse_voice_server_update(self, data) -> None:
...
+    def parse_voice_channel_status_update(self, data) -> None:
+        guild = self._get_guild(int(data["guild_id"]))
+        channel_id = int(data["id"])
+        if guild is not None:
+            channel = guild.get_channel(channel_id)
+            if channel is not None:
+                old_status = channel.status
+                channel.status = data.get("status", None)
+                self.dispatch(
+                    "voice_channel_status_update", channel, old_status, channel.status
```
Comment (plun1331) [inline discord/state.py:1797]: Consider adding a `raw_voice_channel_status_update` event

**Review by plun1331 — CHANGES_REQUESTED** [review]

File: discord/state.py (hunk 4 above)
Comment (Lulalaby) [reply discord/state.py:1797]: yeah, big bots want it

**Review by Lulalaby — DISMISSED** [review]

**Comment by Icebluewolf** [comment]
I fixed the arg names and tested all of the changes Plun and Lala made. 👍

**Review by Lulalaby — APPROVED** [review]

**Review by Dorukyum — APPROVED** [review]
Thank you

**Comment by Lulalaby** [comment]
We need the audit log events in another pull request to complete the voice channel status feature implementation

**Pull request merged**